src/meetingreadinesschecker/DefaultMeetingReadinessChecker.ts

```diff
@@ -0,0 +1,451 @@
+        return CheckCameraResolutionFeedback.PermissionDenied;
+      }
+      return CheckCameraResolutionFeedback.Failure;
+    } finally {
+      if (stream) {
+        stream.getTracks().forEach(function(track) {
+          track.stop();
+        });
+      }
+    }
+    return CheckCameraResolutionFeedback.Success;
+  }
+
+  async checkContentShareConnectivity(): Promise<CheckContentShareConnectivityFeedback> {
+    let isContentShareStarted = false;
+    let isAudioVideoStarted = false;
+
+    const contentShareObserver: ContentShareObserver = {
+      contentShareDidStart: () => {
+        isContentShareStarted = true;
+      },
+    };
+    const observer: AudioVideoObserver = {
+      audioVideoDidStart: () => {
+        isAudioVideoStarted = true;
+      },
+    };
+
+    try {
+      this.meetingSession.audioVideo.addObserver(observer);
+      this.meetingSession.audioVideo.start();
+
+      this.meetingSession.audioVideo.addContentShareObserver(contentShareObserver);
+      await this.meetingSession.audioVideo.startContentShareFromScreenCapture();
+
+      await this.executeTimeoutTask(async () => {
+        return isAudioVideoStarted && isContentShareStarted;
+      });
+
+      this.meetingSession.audioVideo.removeObserver(observer);
+      if (!isAudioVideoStarted) {
+        return CheckContentShareConnectivityFeedback.ConnectionFailure;
+      }
+      await this.stopMeeting();
+      return isContentShareStarted
+        ? CheckContentShareConnectivityFeedback.Success
+        : CheckContentShareConnectivityFeedback.Timeout;
+    } catch (error) {
+      this.logger.error(`MeetingReadinessChecker: Content share check failed with error ${error}`);
+      if (error.name === 'NotAllowedError') {
+        return CheckContentShareConnectivityFeedback.PermissionDenied;
+      } else {
+        return CheckContentShareConnectivityFeedback.Failure;
+      }
+    } finally {
+      this.meetingSession.audioVideo.stopContentShare();
+      this.meetingSession.audioVideo.removeContentShareObserver(contentShareObserver);
+    }
+  }
+
+  async checkAudioConnectivity(
+    audioInputDeviceInfo: MediaDeviceInfo
+  ): Promise<CheckAudioConnectivityFeedback> {
+    let audioPresence = false;
+    const audioVideo = this.meetingSession.audioVideo;
+    const attendeePresenceHandler = (
+      attendeeId: string,
+      present: boolean,
+      _externalUserId: string,
+      _dropped: boolean
+    ): void => {
+      if (attendeeId === this.meetingSession.configuration.credentials.attendeeId && present) {
+        audioPresence = true;
+      }
+    };
+    try {
+      const permissionResult = await audioVideo.chooseAudioInputDevice(audioInputDeviceInfo);
+      if (
+        permissionResult === DevicePermission.PermissionDeniedByBrowser ||
+        permissionResult === DevicePermission.PermissionDeniedByUser
+      ) {
+        return CheckAudioConnectivityFeedback.AudioInputPermissionDenied;
```

Comment:
Note that you don't choose `null` here while choosing `null` in the `checkAudioInput` checker API. I think you can remove it `checkAudioInput` checker API too, since that can fail too.